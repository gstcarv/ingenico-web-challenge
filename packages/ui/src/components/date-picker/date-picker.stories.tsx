import type { Meta, StoryObj } from "@storybook/react-vite";
import { cn } from "../../utils";
import { Button } from "../button";
import { DatePicker } from "./date-picker";

const meta: Meta<typeof DatePicker> = {
    title: "Components/DatePicker",
    component: DatePicker,
    parameters: {
        layout: "centered",
    },
    tags: ["autodocs"],
    argTypes: {
        value: {
            control: "date",
        },
        onChange: {
            action: "date changed",
        },
        placeholder: {
            control: "text",
        },
        format: {
            control: "text",
        },
    },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        placeholder: "Select date...",
    },
};

export const WithValue: Story = {
    args: {
        value: new Date("2024-01-15"),
        placeholder: "Select date...",
    },
};

export const WithPrefix: Story = {
    args: {
        prefix: "ðŸ“…",
        placeholder: "Select date...",
    },
};

export const CustomFormat: Story = {
    args: {
        format: "DD/MM/YYYY",
        placeholder: "Select date...",
    },
};

export const Disabled: Story = {
    args: {
        disabled: true,
        placeholder: "Select date...",
    },
};

export const WithCustomRender: Story = {
    args: {
        placeholder: "Select date...",
        renderInput: ({ value, placeholder, onClick, onKeyDown, disabled }) => (
            <Button
                variant="outline"
                onClick={onClick}
                onKeyDown={onKeyDown}
                disabled={disabled}
                className="w-[300px] justify-between"
                role="button"
                aria-haspopup="dialog"
                aria-expanded={false}
                aria-label="Date picker"
            >
                <span className={cn(value ? "text-neutral-900" : "text-neutral-500", "block w-full text-center")}>
                    {value || placeholder}
                </span>
            </Button>
        ),
    },
};
