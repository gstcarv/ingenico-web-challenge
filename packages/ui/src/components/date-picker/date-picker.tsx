import { cva } from "class-variance-authority";
import dayjs from "dayjs";
import { Calendar } from "lucide-react";
import { ComponentProps, useRef, useState } from "react";
import { DayPicker } from "react-day-picker";
import "react-day-picker/style.css";
import { useClickOutside } from "../../hooks/useClickOutside";
import { cn } from "../../utils";
import { InputBase, InputBaseProps } from "../input-base";

export type DatePickerRenderInputProps = {
    value: string;
    placeholder: string;
    onClick: () => void;
    onKeyDown: (e: React.KeyboardEvent) => void;
    disabled?: boolean;
    "data-testid"?: string;
};

export type DatePickerProps = Omit<ComponentProps<typeof DayPicker>, "mode" | "selected" | "onSelect"> &
    Omit<InputBaseProps, "onClick" | "onKeyDown"> & {
        value?: Date;
        onChange?: (date: Date | undefined) => void;
        placeholder?: string;
        format?: string;
        disabled?: boolean;
        "data-testid"?: string;
        renderInput?: (props: DatePickerRenderInputProps) => React.ReactNode;
    };

const datePickerVariants = cva(
    "absolute top-full left-0 mt-1 bg-white border border-neutral-200 rounded-lg shadow-lg z-50 transition-all duration-200 ease-out",
    {
        variants: {
            variant: {
                default: "",
            },
        },
        defaultVariants: {
            variant: "default",
        },
    },
);

export const DatePicker = ({
    className,
    prefix,
    suffix,
    value,
    onChange,
    placeholder = "Select date...",
    format = "DD/MM/YYYY",
    disabled,
    "data-testid": dataTestId,
    renderInput,
    ...props
}: DatePickerProps) => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedDate, setSelectedDate] = useState<Date | undefined>(value);
    const containerRef = useRef<HTMLDivElement>(null);

    const handleDateSelect = (date: Date | undefined) => {
        setSelectedDate(date);
        onChange?.(date);
        setIsOpen(false);
    };

    // Handle click outside to close popover
    useClickOutside(containerRef, () => setIsOpen(false), isOpen);

    const handleInputClick = () => {
        setIsOpen(!isOpen);
    };

    const handleInputKeyDown = (e: React.KeyboardEvent) => {
        if (e.key === "Enter" || e.key === " ") {
            e.preventDefault();
            setIsOpen(!isOpen);
        }
        if (e.key === "Escape") {
            setIsOpen(false);
        }
    };

    const displayValue = selectedDate ? dayjs(selectedDate).format(format) : "";

    const inputProps = {
        value: displayValue,
        placeholder,
        onClick: handleInputClick,
        onKeyDown: handleInputKeyDown,
        disabled,
        "data-testid": dataTestId,
    };

    return (
        <div className="relative" data-testid={dataTestId} ref={containerRef}>
            {renderInput ? (
                renderInput(inputProps)
            ) : (
                <InputBase
                    className={className}
                    prefix={prefix}
                    suffix={suffix || <Calendar size={20} className="text-neutral-500" />}
                    onClick={handleInputClick}
                    onKeyDown={handleInputKeyDown}
                    tabIndex={0}
                    role="button"
                    aria-haspopup="dialog"
                    aria-expanded={isOpen}
                    aria-label={props["aria-label"] || "Date picker"}
                    aria-describedby={props["aria-describedby"]}
                    aria-disabled={disabled}
                >
                    <input
                        type="text"
                        value={displayValue}
                        placeholder={placeholder}
                        readOnly
                        className={cn(
                            "w-full h-full bg-transparent focus-visible:ring-2 focus-visible:ring-offset focus-visible:ring-brand-primary rounded-lg outline-none cursor-pointer",
                            prefix ? "pl-12" : "px-4",
                        )}
                    />
                </InputBase>
            )}

            <div
                className={cn(
                    datePickerVariants(),
                    {
                        "opacity-100 scale-100 translate-y-0": isOpen,
                        "opacity-0 scale-95 -translate-y-2 pointer-events-none": !isOpen,
                    },
                    className,
                )}
            >
                <DayPicker
                    mode="single"
                    selected={selectedDate}
                    showOutsideDays
                    className="p-6"
                    {...props}
                    onSelect={(d) => handleDateSelect(d)}
                />
            </div>
        </div>
    );
};
