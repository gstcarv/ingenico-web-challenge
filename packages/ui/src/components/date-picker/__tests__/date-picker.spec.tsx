import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { DatePicker, DatePickerRenderInputProps } from "../date-picker";

describe("DatePicker", () => {
    it("renders date picker input with correct attributes", () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        expect(datePicker).toBeInTheDocument();

        const inputBase = datePicker.querySelector('[role="button"]');
        expect(inputBase).toHaveAttribute("aria-haspopup", "dialog");
        expect(inputBase).toHaveAttribute("aria-expanded", "false");
    });

    it("renders with default placeholder", () => {
        render(<DatePicker data-testid="date-picker" />);

        const input = screen.getByPlaceholderText("Select date...");
        expect(input).toBeInTheDocument();
    });

    it("renders with custom placeholder", () => {
        render(<DatePicker placeholder="Choose a date" data-testid="date-picker" />);

        const input = screen.getByPlaceholderText("Choose a date");
        expect(input).toBeInTheDocument();
    });

    it("renders calendar icon as default suffix", () => {
        render(<DatePicker data-testid="date-picker" />);

        const calendarIcon = screen.getByTestId("date-picker").querySelector("svg");
        expect(calendarIcon).toBeInTheDocument();
    });

    it("renders custom suffix when provided", () => {
        const CustomSuffix = () => <span data-testid="custom-suffix">ğŸ“…</span>;

        render(<DatePicker suffix={<CustomSuffix />} data-testid="date-picker" />);

        expect(screen.getByTestId("custom-suffix")).toBeInTheDocument();
    });

    it("renders prefix when provided", () => {
        render(<DatePicker prefix="ğŸ“…" data-testid="date-picker" />);

        expect(screen.getByText("ğŸ“…")).toBeInTheDocument();
    });

    it("opens calendar on click", async () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        fireEvent.click(inputBase!);

        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });
    });

    it("opens calendar on Enter key", async () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        fireEvent.keyDown(inputBase!, { key: "Enter" });

        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });
    });

    it("opens calendar on Space key", async () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        fireEvent.keyDown(inputBase!, { key: " " });

        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });
    });

    it("closes calendar on Escape key", async () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');

        // Open calendar first
        fireEvent.click(inputBase!);
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });

        // Close with Escape
        fireEvent.keyDown(inputBase!, { key: "Escape" });
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "false");
        });
    });

    it("closes calendar when clicking outside", async () => {
        render(
            <div>
                <div data-testid="outside-element">Outside content</div>
                <DatePicker data-testid="date-picker" />
            </div>,
        );

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        const outsideElement = screen.getByTestId("outside-element");

        // Open calendar first
        fireEvent.click(inputBase!);
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });

        // Click outside to close
        fireEvent.mouseDown(outsideElement);
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "false");
        });
    });

    it("does not close calendar when clicking inside", async () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');

        // Open calendar first
        fireEvent.click(inputBase!);
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });

        // Click inside the date picker container
        fireEvent.mouseDown(datePicker);
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });
    });

    it("displays selected date in correct format", () => {
        // Use a date that's clearly January 15th regardless of timezone
        const testDate = new Date(2024, 0, 15, 12, 0, 0); // January 15, 2024 at 12:00 PM
        render(<DatePicker value={testDate} data-testid="date-picker" />);

        const input = screen.getByDisplayValue("15/01/2024");
        expect(input).toBeInTheDocument();
    });

    it("displays selected date in custom format", () => {
        // Use a date that's clearly January 15th regardless of timezone
        const testDate = new Date(2024, 0, 15, 12, 0, 0); // January 15, 2024 at 12:00 PM
        render(<DatePicker value={testDate} format="DD/MM/YYYY" data-testid="date-picker" />);

        const input = screen.getByDisplayValue("15/01/2024");
        expect(input).toBeInTheDocument();
    });

    it("calls onChange when date is selected", async () => {
        const handleChange = vi.fn();
        render(<DatePicker onChange={handleChange} data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        fireEvent.click(inputBase!);

        // Wait for calendar to open and find a day to click
        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });

        // Find and click a day (this is a simplified test)
        const dayButtons = screen
            .getAllByRole("button")
            .filter((button) => button.getAttribute("aria-label")?.includes("day"));

        if (dayButtons.length > 0) {
            fireEvent.click(dayButtons[0]);
            expect(handleChange).toHaveBeenCalled();
        }
    });

    it("closes calendar after date selection", async () => {
        const handleChange = vi.fn();
        render(<DatePicker onChange={handleChange} data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        fireEvent.click(inputBase!);

        await waitFor(() => {
            expect(inputBase).toHaveAttribute("aria-expanded", "true");
        });

        // Find and click a day
        const dayButtons = screen
            .getAllByRole("button")
            .filter((button) => button.getAttribute("aria-label")?.includes("day"));

        if (dayButtons.length > 0) {
            fireEvent.click(dayButtons[0]);

            await waitFor(() => {
                expect(inputBase).toHaveAttribute("aria-expanded", "false");
            });
        }
    });

    it("applies custom className", () => {
        const { container } = render(<DatePicker className="custom-class" data-testid="date-picker" />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it("handles disabled state", () => {
        render(<DatePicker disabled data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        expect(inputBase).toHaveAttribute("aria-disabled", "true");
    });

    it("handles empty value", () => {
        render(<DatePicker data-testid="date-picker" />);

        const input = screen.getByPlaceholderText("Select date...");
        expect(input).toHaveValue("");
    });

    it("forwards all HTML attributes", () => {
        render(<DatePicker data-testid="date-picker" aria-label="Select birth date" />);

        const datePicker = screen.getByTestId("date-picker");
        expect(datePicker).toHaveAttribute("data-testid", "date-picker");
    });
});

describe("DatePicker Accessibility", () => {
    it("supports aria-label", () => {
        render(<DatePicker aria-label="Select date" data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        expect(inputBase).toHaveAttribute("aria-label", "Select date");
    });

    it("supports aria-describedby", () => {
        render(<DatePicker aria-describedby="description" data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        expect(inputBase).toHaveAttribute("aria-describedby", "description");
    });

    it("has proper keyboard navigation support", () => {
        render(<DatePicker data-testid="date-picker" />);

        const datePicker = screen.getByTestId("date-picker");
        const inputBase = datePicker.querySelector('[role="button"]');
        expect(inputBase).toHaveAttribute("tabIndex", "0");
    });

    it("renders custom input when renderInput is provided", () => {
        const CustomInput = ({ value, placeholder, onClick, onKeyDown, disabled }: DatePickerRenderInputProps) => (
            <button
                onClick={onClick}
                onKeyDown={onKeyDown}
                disabled={disabled}
                data-testid="custom-input"
                className="custom-button"
            >
                {value || placeholder}
            </button>
        );

        render(<DatePicker data-testid="date-picker" renderInput={CustomInput} placeholder="Custom placeholder" />);

        const customInput = screen.getByTestId("custom-input");
        expect(customInput).toBeInTheDocument();
        expect(customInput).toHaveTextContent("Custom placeholder");
        expect(customInput).toHaveClass("custom-button");
    });
});

describe("DatePicker Integration", () => {
    it("works with form elements", () => {
        const handleSubmit = vi.fn((e) => e.preventDefault());

        render(
            <form onSubmit={handleSubmit}>
                <DatePicker data-testid="date-picker" />
                <button type="submit">Submit</button>
            </form>,
        );

        const datePicker = screen.getByTestId("date-picker");
        const submitButton = screen.getByText("Submit");

        expect(datePicker).toBeInTheDocument();
        expect(submitButton).toBeInTheDocument();
    });
});
