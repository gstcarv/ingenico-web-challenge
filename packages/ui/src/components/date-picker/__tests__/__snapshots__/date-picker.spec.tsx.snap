// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatePicker > applies custom className 1`] = `
<div
  class="relative"
  data-testid="date-picker"
>
  <div
    aria-expanded="false"
    aria-haspopup="dialog"
    aria-label="Date picker"
    class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium custom-class"
    role="button"
    tabindex="0"
  >
    <input
      class="w-full h-full bg-transparent focus-visible:ring-2 focus-visible:ring-offset focus-visible:ring-brand-primary rounded-lg outline-none cursor-pointer px-4"
      placeholder="Select date..."
      readonly=""
      type="text"
      value=""
    />
    <div
      class="absolute right-4 top-1/2 transform -translate-y-1/2 text-neutral-500"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-calendar text-neutral-500"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 2v4"
        />
        <path
          d="M16 2v4"
        />
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="4"
        />
        <path
          d="M3 10h18"
        />
      </svg>
    </div>
  </div>
  <div
    class="absolute top-full left-0 mt-1 bg-white border border-neutral-200 rounded-lg shadow-lg z-50 transition-all duration-200 ease-out opacity-0 scale-95 -translate-y-2 pointer-events-none custom-class"
  >
    <div
      class="rdp-root p-6"
      data-mode="single"
    >
      <div
        class="rdp-months"
      >
        <nav
          aria-label=""
          class="rdp-nav"
        >
          <button
            aria-label="Go to the Previous Month"
            class="rdp-button_previous"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
              />
            </svg>
          </button>
          <button
            aria-label="Go to the Next Month"
            class="rdp-button_next"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="8 18.112 14.18888889 12 8 5.87733333 9.91111111 4 18 12 9.91111111 20"
              />
            </svg>
          </button>
        </nav>
        <div
          class="rdp-month"
        >
          <div
            class="rdp-month_caption"
          >
            <span
              aria-live="polite"
              class="rdp-caption_label"
              role="status"
            >
              August 2025
            </span>
          </div>
          <table
            aria-label="August 2025"
            aria-multiselectable="false"
            class="rdp-month_grid"
            role="grid"
          >
            <thead
              aria-hidden="true"
            >
              <tr
                class="rdp-weekdays"
              >
                <th
                  aria-label="Sunday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Su
                </th>
                <th
                  aria-label="Monday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Mo
                </th>
                <th
                  aria-label="Tuesday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Tu
                </th>
                <th
                  aria-label="Wednesday"
                  class="rdp-weekday"
                  scope="col"
                >
                  We
                </th>
                <th
                  aria-label="Thursday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Th
                </th>
                <th
                  aria-label="Friday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Fr
                </th>
                <th
                  aria-label="Saturday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Sa
                </th>
              </tr>
            </thead>
            <tbody
              class="rdp-weeks"
            >
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-07-27"
                  data-month="2025-07"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, July 27th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    27
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-07-28"
                  data-month="2025-07"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, July 28th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    28
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-07-29"
                  data-month="2025-07"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, July 29th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    29
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-07-30"
                  data-month="2025-07"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, July 30th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    30
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-07-31"
                  data-month="2025-07"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, July 31st, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    31
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-01"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, August 1st, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    1
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-02"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, August 2nd, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    2
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2025-08-03"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, August 3rd, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    3
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-04"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, August 4th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    4
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-05"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, August 5th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    5
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-06"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, August 6th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    6
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-07"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, August 7th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    7
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-08"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, August 8th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    8
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-09"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, August 9th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    9
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2025-08-10"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, August 10th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    10
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-11"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, August 11th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    11
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-12"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, August 12th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    12
                  </button>
                </td>
                <td
                  class="rdp-day rdp-today"
                  data-day="2025-08-13"
                  data-today="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Today, Wednesday, August 13th, 2025"
                    class="rdp-day_button"
                    tabindex="0"
                    type="button"
                  >
                    13
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-14"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, August 14th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    14
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-15"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, August 15th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    15
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-16"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, August 16th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    16
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2025-08-17"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, August 17th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    17
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-18"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, August 18th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    18
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-19"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, August 19th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    19
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-20"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, August 20th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    20
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-21"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, August 21st, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    21
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-22"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, August 22nd, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    22
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-23"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, August 23rd, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    23
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2025-08-24"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, August 24th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    24
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-25"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, August 25th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    25
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-26"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, August 26th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    26
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-27"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, August 27th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    27
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-28"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, August 28th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    28
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-29"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, August 29th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    29
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2025-08-30"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, August 30th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    30
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2025-08-31"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, August 31st, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    31
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-09-01"
                  data-month="2025-09"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, September 1st, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    1
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-09-02"
                  data-month="2025-09"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, September 2nd, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    2
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-09-03"
                  data-month="2025-09"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, September 3rd, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    3
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-09-04"
                  data-month="2025-09"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, September 4th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    4
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-09-05"
                  data-month="2025-09"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, September 5th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    5
                  </button>
                </td>
                <td
                  class="rdp-day rdp-outside"
                  data-day="2025-09-06"
                  data-month="2025-09"
                  data-outside="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, September 6th, 2025"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    6
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
